ibis/expr/types/json.py

```diff
@@ -17,10 +18,114 @@ class JSONValue(Value):
     def __getitem__(
         self, key: str | int | ir.StringValue | ir.IntegerValue
     ) -> JSONValue:
+        """Access an JSON object's value or JSON array's element at `key`.
+
+        Parameters
+        ----------
+        key
+            Object field name or integer array index
+
+        Returns
+        -------
+        JSONValue
+            Element located at `key`
+
+        Examples
+        --------
+        Construct a table with a JSON column
+
+        >>> import json, ibis
+        >>> ibis.options.interactive = True
+        >>> rows = [{"js": json.dumps({"a": [i, 1]})} for i in range(2)]
+        >>> t = ibis.memtable(rows, schema=ibis.schema(js="json"))
+        >>> t
+        ┏━━━━━━━━━━━━━━━┓
+        ┃ js            ┃
+        ┡━━━━━━━━━━━━━━━┩
+        │ json          │
+        ├───────────────┤
+        │ {'a': [0, 1]} │
+        │ {'a': [1, 1]} │
+        └───────────────┘
+
+        Extract the `"a"` field
+
+        >>> t.js["a"]
+        ┏━━━━━━━━━━━━━━━━━━━━━━┓
+        ┃ JSONGetItem(js, 'a') ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━━┩
+        │ json                 │
+        ├──────────────────────┤
+        │ [0, 1]               │
+        │ [1, 1]               │
+        └──────────────────────┘
+
+        Extract the first element of the JSON array at `"a"`
+
+        >>> t.js["a"][0]
+        ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+        ┃ JSONGetItem(JSONGetItem(js, 'a'), 0) ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
+        │ json                                 │
+        ├──────────────────────────────────────┤
+        │ 0                                    │
+        │ 1                                    │
+        └──────────────────────────────────────┘
+
+        Extract a non-existent field
+
+        >>> t.js.a["foo"]
+        ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+        ┃ JSONGetItem(JSONGetItem(js, 'a'), 'foo') ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
+        │ json                                     │
+        ├──────────────────────────────────────────┤
+        │ ∅                                        │
+        │ ∅                                        │
+        └──────────────────────────────────────────┘
+
+        Try to extract an array element, returns `NULL`
+
+        >>> t.js[20]
+        ┏━━━━━━━━━━━━━━━━━━━━━┓
+        ┃ JSONGetItem(js, 20) ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━┩
+        │ json                │
+        ├─────────────────────┤
+        │ ∅                   │
+        │ ∅                   │
+        └─────────────────────┘
+        """
         import ibis.expr.operations as ops
 
         return ops.JSONGetItem(self, key).to_expr()
 
+    @property
+    def map(self) -> ir.MapValue:
+        """Cast this JSON blob to a map of string to JSON.
+
+        Use this property to unlock map functionality on JSON objects.
+
+        Returns
+        -------
+        MapValue
+            Map of string to JSON
```

Comment:
I'd like to add examples here, but these only work with snowflake for the time being.